src/imports.rs

```diff
@@ -988,7 +1000,12 @@ mod test {
         test_merge!(["a::b::{c, d}", "a::b::{e, f}"], ["a::b::{c, d, e, f}"]);
         test_merge!(["a::b::c", "a::b"], ["a::{b, b::c}"]);
         test_merge!(["a::b", "a::b"], ["a::b"]);
-        test_merge!(["a", "a::b", "a::b::c"], ["a::{self, b::{self, c}}"]);
+        test_merge!(["a", "a::b", "a::b::c"], ["a::{self, b, b::c}"]);
```

Comment:
Hmm, looking back to this test, merging `use a;` into `use a::{self};` does not seem to be correct w.r.t. #3750.
@valff IIUC this has existed before this PR, but while you are at it, would you mind having a look if you have time? This won't be a block for this PR - if this is not easy to fix then we can leave it for the future work.

Reply:
@topecongiro I have kept `use a::{self}` as a special case where `self` is at depth 2, because the first path segment is always a crate name or a special keyword (at least in edition 2018?). So if the first segment is always a module (crate), it should be correct w.r.t. #3750. I find this special case useful because this way we still have a single `use` statement per crate.

Reply:
Ah, I misunderstood this. Thanks!